### PR TITLE
✨ Unit tests for re-usable workflow call

### PR DESCRIPTION
**Pull request by laurentsimon**
Note: the actionlinter is pretty strict: if env variables are declared, or runs-on id declared, it will fail.. which plays in our favor

### CODE REVIEW

**Comment by laurentsimon** [comment]
ready for review. Any comments?

File: provenance-verifier/pkg/workflow.go [hunk 1]
```diff
@@ -119,6 +122,23 @@ func (w *Workflow) validateJobSteps(job *actionlint.Job) error {
...
+	if job == nil || job.WorkflowCall == nil || job.WorkflowCall.Uses == nil {
+		return false, nil
+	}
+
+	values := strings.Split(job.WorkflowCall.Uses.Value, "@")
+	if len(values) != 2 {
+		return false, fmt.Errorf("%s: %s: %w", getJobIdentity(job),
+			job.WorkflowCall.Uses.Value, errorInvalidReUsableWorkflow)
+	}
+	fmt.Println(values)
```
Comment (asraa) [inline provenance-verifier/pkg/workflow.go:136]: debugging?

File: provenance-verifier/pkg/workflow.go [hunk 2]
```diff
@@ -119,6 +122,23 @@ func (w *Workflow) validateJobSteps(job *actionlint.Job) error {
 	return nil
 }
 
+// =============== Re-usable workflow ================ //
+func (w *Workflow) IsJobCallingTrustedReusableWorkflow(job *actionlint.Job) (bool, error) {
+	if job == nil || job.WorkflowCall == nil || job.WorkflowCall.Uses == nil {
+		return false, nil
+	}
+
+	values := strings.Split(job.WorkflowCall.Uses.Value, "@")
```
Comment (asraa) [inline provenance-verifier/pkg/workflow.go:131]: should we include that the error may be that it's not pinned (assuming that if no tag is mentioned it defaults to main)

File: provenance-verifier/pkg/workflow.go (hunk 1 above)
Comment (laurentsimon) [reply provenance-verifier/pkg/workflow.go:136]: goo catch, will remove

File: provenance-verifier/pkg/workflow.go (hunk 2 above)
Comment (laurentsimon) [reply provenance-verifier/pkg/workflow.go:131]: ill add a better message.
Do we want to enforce pinning by hash? Currently it accepts pinning by version (easier for debugging), but we may want to make this a hash.
I'll create a tracking issue for this.

File: provenance-verifier/pkg/workflow.go (hunk 1 above)
Comment (laurentsimon) [reply provenance-verifier/pkg/workflow.go]: done.

**Review by asraa — APPROVED** [review]

**Pull request merged**